File: epikjjh/baekjoon/14501.py
import sys
input = sys.stdin.readline

n = int(input().rstrip())
time = [0]*n
price = [0]*n
for i in range(n):
    time_elem, price_elem = map(int, input().rstrip().split())
    time[i] = time_elem
    price[i] = price_elem

    
def find_max(cur_day, cur_price):
    if cur_day >= n:
        return cur_price
    if cur_day+time[cur_day] > n:
        cur_price = find_max(cur_day+1, cur_price)
    else:    
        cur_price = max(find_max(cur_day+1, cur_price), find_max(cur_day+time[cur_day], cur_price+price[cur_day]))
    return cur_price


print(find_max(0,0))

File: epikjjh/baekjoon/14501.go
package main

import "fmt"

var n int
var time, price []int

func max(x, y int) int{
    if x < y {
        return y
    }
    return x
}
func findMax(cur_day, cur_price int) int{
    if cur_day >= n{
        return cur_price
    }
    if cur_day+time[cur_day] > n{
        return findMax(cur_day+1,cur_price)
    }else{
        return max(findMax(cur_day+1,cur_price), findMax(cur_day+time[cur_day],cur_price+price[cur_day]))
    }
}
func main(){
    fmt.Scan(&n)
    time = make([]int,0,n)
    price = make([]int,0,n)
    for i:=1;i<=n;i++{
        var time_elem, price_elem int
        fmt.Scanln(&time_elem, &price_elem)
        time = append(time, time_elem)
        price = append(price, price_elem)
    }
    fmt.Println(findMax(0,0))
}